lib/Models/ArcGisMapServerCatalogItem.js

```diff
@@ -84,7 +86,13 @@ var ArcGisMapServerCatalogItem = function(terria) {
      */
     this.allowFeaturePicking = true;
 
-    knockout.track(this, ['layers', 'maximumScale', '_legendUrl', '_generatedLegendUrl', 'maximumScaleBeforeMessage', 'showTilesAfterMessage', 'allowFeaturePicking']);
+    /**
+     * Gets or sets the URL to use for requesting tokens.
+     * @type {String}
+     */
+    this.tokenSource = undefined;
```

Comment:
I think `tokenUrl` would be a better name than `tokenSource`.

Reply:
Thanks much better.

8c83029b95bfccb2748606190c2018035b1bf6ed